packages/js/components/src/experimental-select-tree-control/select-tree.tsx

```diff
@@ -86,18 +93,22 @@ export const SelectTree = function SelectTree( {
 
 	const selectedItemsFocusHandle = useRef< SelectedItemFocusHandle >( null );
 
-	function isEventOutside( event: React.FocusEvent ) {
+	function isEventOutside( event: React.SyntheticEvent | React.FocusEvent ) {
+		let target: Element | null = event.currentTarget;
+		if ( isBlurEvent( event ) ) {
+			target = event.relatedTarget;
+		}
```

Comment:
Needed this, as the above is called on blur and on focus. The blur event has the `FocusEvent` and thus includes the `relatedTarget`. The focus event only returns `SyntheticEvent` and thus does not have `relatedTarget`, to be specific the `onFocusOutside` -> https://github.com/WordPress/gutenberg/blob/775eed2cb8e480e5978fd888ba8df7a98e7e53f4/packages/components/src/popover/types.ts#L95

Reply:
Hmm yeah that does work and I updated it: https://github.com/woocommerce/woocommerce/pull/54197/commits/218601a3384a8871c281a97114febd9e60cbdba0
but the one downside is that it doesn't auto select the type because of this. So I still need the `isBlurEvent` function using the `is`. I made use of `type` within there.

Reply:
Yeah, type guards for the win!